webapp/app/components/common/modal/MsModal.jsx

```diff
@@ -4,29 +4,52 @@ import PropTypes from 'prop-types';
 
 class MsModal extends React.PureComponent {
   static propTypes= {
-    okButtonLabel: PropTypes.string.isRequired,
-    cancelButtonLabel: PropTypes.string.isRequired,
+    okButtonLabel: PropTypes.string,
+    cancelButtonLabel: PropTypes.string,
     body: PropTypes.any.isRequired,
     isOpen: PropTypes.bool.isRequired,
-    okCallback: PropTypes.func.isRequired,
-    cancelCallback: PropTypes.func.isRequired,
-    modalTitle: PropTypes.string.isRequired
+    okCallback: PropTypes.func,
+    cancelCallback: PropTypes.func,
+    modalTitle: PropTypes.string.isRequired,
+    showFooter: PropTypes.bool
+  };
+
+  static defaultProps = {
+    showFooter: true
   };
 
   render() {
-    const {isOpen, cancelCallback, modalTitle, body, okCallback, okButtonLabel, cancelButtonLabel} = this.props;
+    const {isOpen, cancelCallback, modalTitle, body } = this.props;
     return (
       <Modal isOpen={isOpen} toggle={cancelCallback}>
         <ModalHeader toggle={cancelCallback}>{modalTitle}</ModalHeader>
         <ModalBody>
           {body}
         </ModalBody>
+        {this.renderFooter()}
+      </Modal>
+    );
+  }
+
+  renderFooter() {
+    const {
+      cancelCallback,
+      cancelButtonLabel,
+      okCallback,
+      okButtonLabel
+    } = this.props;
+    let dataToRender = null;
+
+    if (this.props.showFooter) {
```

Comment:
Could not find the added test case for this newly added if branch, this will lower our branches coverage.

We should make sure to test all newly intoduced modifications and functionality, and as a rule of thumb always bring our coverage up or at least keep it the same after modifications or additions.

We should add the corresponding test, thanks

Reply:
Done!